autolab/Autolab
# Don't mark error-initialized scores as graded by autograder

## Description
- Set `grader_id` of error-initialized scores to `-1`

## Motivation and Context
Currently, if there is an error parsing the score json of the autograder's output, all problems are given a score of 0 and marked as graded by the autograder. This has the effect of preventing an instructor from annotating problems that were not meant to be autograded.

Some context
- The `grader_id` of a score records who graded a particular problem
  - The autograder has a `grader_id` of 0
  - Other users have a `grader_id` of > 0
- To prevent annotations on autograded problems, there are checks for `grader_id == 0`
- Currently, on error, we set the `grader_id` of all ungraded problems to 0, which effectively marks them as autograded by the autograder
  - The idea is, we do not know which problems were intended to be autograded, so we "autograde" everything and set a feedback so that a user can click on the score and view the error message
  - In this PR, we set the `grader_id` to -1 instead, so that our checks do not treat the scores as coming from the autograder, allowing us to annotate them normally

Note: In the future, if we get rid of the gradesheet / view feedback workflow altogether, we must be mindful about displaying error messages somewhere on view submission. The autograder output does not show it.

Closes #1641

## How Has This Been Tested?
- Create an assessment whose autograder does not output a score json at the end (e.g. install `hello.tar`, modify `src/driver.sh`, run `make`)
- Make a submission, observe that all scores are "clickable" and bring you to the autograder output. This is the same behavior as before -- it allows users to see the error message
- Ensure that you can now make an annotation for all problems (including the one that was meant to be autograded, since we do not make a distinction)
- As a sanity check, ensure that when the autograder outputs a score json normally, only the autograded score is clickable, we can only make annotations for the non-autograded problems, etc.

**All scores are clickable -- same as before**
![Screenshot 2022-11-12 at 19 12 54](https://user-images.githubusercontent.com/9074856/201499610-15d13278-b80d-4579-aed5-bc37bdbabf5b.png)

**Viewable error message -- same as before**
![Screenshot 2022-11-12 at 19 14 54](https://user-images.githubusercontent.com/9074856/201499668-401164d9-ed2d-4838-bf36-268cf5e1cd13.png)

**Annotations can now be made -- new**
![Screenshot 2022-11-12 at 19 15 01](https://user-images.githubusercontent.com/9074856/201499665-49ddae9f-9f30-46cd-9496-5fc78f59b66c.png)

Also, "Graded By" now shows something if the score came from the autograder / error handler

![Screenshot 2022-11-13 at 11 19 18](https://user-images.githubusercontent.com/9074856/201532622-e320c25c-5654-4eda-9092-1a6217b581b8.png)
![Screenshot 2022-11-13 at 11 19 30](https://user-images.githubusercontent.com/9074856/201532623-a420d373-b722-404e-b919-1e3cb8e631cb.png)


## Types of changes

## Checklist:
